### PR TITLE
[OpenBLAS] Add v0.3.20

### PR DESCRIPTION
**Pull request by giordano**

### CODE REVIEW

File: O/OpenBLAS/OpenBLAS@0.3.20/bundled/patches/neoverse-generic-kernels.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
+diff --git a/kernel/arm64/KERNEL.NEOVERSEN1 b/kernel/arm64/KERNEL.NEOVERSEN1
```
Comment (giordano) [inline O/OpenBLAS/OpenBLAS@0.3.20/bundled/patches/neoverse-generic-kernels.patch:1]: @Keno: https://github.com/xianyi/OpenBLAS/pull/3052 supposedly fixed the issue you reported at https://github.com/xianyi/OpenBLAS/issues/2998, did you ever get a chance to verify whether we can drop this patch now?

**Comment by giordano** [comment]
```
[17:11:19] ../libopenblas64_p-r0.3.20.a(dgemm_small_kernel_nn_SKYLAKEX.o): In function `dgemm_small_kernel_nn_SKYLAKEX':
[17:11:19] dgemm_small_kernel_nn_skylakex.c:(.text+0x2ddd): undefined reference to `_mm512_reduce_add_pd'
[17:11:19] dgemm_small_kernel_nn_skylakex.c:(.text+0x2e1c): undefined reference to `_mm512_reduce_add_pd'
[17:11:19] dgemm_small_kernel_nn_skylakex.c:(.text+0x2e8e): undefined reference to `_mm512_reduce_add_pd'
[17:11:19] dgemm_small_kernel_nn_skylakex.c:(.text+0x2eea): undefined reference to `_mm512_reduce_add_pd'
[17:11:19] dgemm_small_kernel_nn_skylakex.c:(.text+0x30c7): undefined reference to `_mm512_reduce_add_pd'
[17:11:19] ../libopenblas64_p-r0.3.20.a(dgemm_small_kernel_nn_SKYLAKEX.o):dgemm_small_kernel_nn_skylakex.c:(.text+0x3106): more undefined references to `_mm512_reduce_add_pd' follow
[17:11:19] ../libopenblas64_p-r0.3.20.a(sgemm_small_kernel_nn_SKYLAKEX.o): In function `sgemm_small_kernel_nn_SKYLAKEX':
[17:11:19] sgemm_small_kernel_nn_skylakex.c:(.text+0x2ca3): undefined reference to `_mm512_reduce_add_ps'
[17:11:19] sgemm_small_kernel_nn_skylakex.c:(.text+0x2cdc): undefined reference to `_mm512_reduce_add_ps'
[17:11:19] sgemm_small_kernel_nn_skylakex.c:(.text+0x2d42): undefined reference to `_mm512_reduce_add_ps'
[17:11:19] sgemm_small_kernel_nn_skylakex.c:(.text+0x2d95): undefined reference to `_mm512_reduce_add_ps'
[17:11:19] sgemm_small_kernel_nn_skylakex.c:(.text+0x2f25): undefined reference to `_mm512_reduce_add_ps'
[17:11:19] ../libopenblas64_p-r0.3.20.a(sgemm_small_kernel_nn_SKYLAKEX.o):sgemm_small_kernel_nn_skylakex.c:(.text+0x2f5e): more undefined references to `_mm512_reduce_add_ps' follow
[17:11:19] collect2: error: ld returned 1 exit status
```

https://github.com/xianyi/OpenBLAS/issues/3530 looks relevant, it appears GCC headers in v6 were wrong.  This wouldn't even be the first time we hit similar issues when building OpenBLAS, see #2142, #2385, #2389, #2415.  Sigh.

File: O/OpenBLAS/OpenBLAS@0.3.20/bundled/patches/neoverse-generic-kernels.patch (hunk 1 above)
Comment (Keno) [reply O/OpenBLAS/OpenBLAS@0.3.20/bundled/patches/neoverse-generic-kernels.patch:1]: No idea

Comment (giordano) [reply O/OpenBLAS/OpenBLAS@0.3.20/bundled/patches/neoverse-generic-kernels.patch:1]: For the record, I verified the patch should now be redundant, as vanilla OpenBLAS now produces the right result: https://github.com/JuliaPackaging/Yggdrasil/pull/7202#issuecomment-1908397135.

**Pull request merged**